lib/v2/screens/explore_screens/explore/explore_screen.dart

```diff
@@ -42,7 +42,10 @@ class ExploreScreen extends StatelessWidget {
                       padding: const EdgeInsets.only(left: 16, right: 16, top: 16),
                       child: ExploreInfoCard(
                         onTap: () {
-                          NavigationService.of(context).navigateTo(Routes.createInvite);
+                          NavigationService.of(context).navigateTo(
```

Comment:
passing blocs around is dangerousss. 
Is there a diff way to do this?

Reply:
The only 2 options I can think are:

- A callback but I don't know how it would work with a named route.
- Wait for the navigation to return a result and based on that make the event from explore.

But why do you say that it is not correct to pass the Bloc?

Reply:
I think we had this conversation before, I honestly dont know if its a good practice or not, but its smells bad, like rotten eggs. 


in this case, You have screen  `Invite` telling screen `Explore` to do things, even when screen `Explore is not even displayed.  What if for some reason Screen `Explore` died? got garbage collected? 

Normally screen `Invite` would let screen `explore` know that something happened and its up to `explore` screen to do something about it or not. 

`Wait for the navigation to return a result and based on that make the event from explore.`

This is the approach i would take.

Reply:
Changed to navigation approach.

However, this implies that for every screen that need to report a previous one, we will have to use WillPopScope in order to report an event.

Also I find it very difficult a screen to die in these cases that a screen is lost like by the garbage collector in any case the whole app would be closed by the OS. Explore is one of the main bottom app screens. I think it applies is when a screen is called from two different screens or if a push and replace is done.